src/gui/editors/AutomationEditor.cpp

```diff
@@ -1626,12 +1626,12 @@ void AutomationEditor::wheelEvent(QWheelEvent * we )
 	else if(abs(we->angleDelta().x()) > abs(we->angleDelta().y())) // scrolling is horizontal
 	{
 		m_leftRightScroll->setValue(m_leftRightScroll->value() -
-							we->angleDelta().x() * 2 / 15);
+							we->angleDelta().x() * 2 / 6 / m_zoomXLevels[m_zoomingXModel.value()]);
 	}
 	else if(we->modifiers() & Qt::ShiftModifier)
 	{
 		m_leftRightScroll->setValue(m_leftRightScroll->value() -
-							we->angleDelta().y() * 2 / 15);
+							we->angleDelta().y() * 2 / 6 / m_zoomXLevels[m_zoomingXModel.value()]);
```

Comment:
There's some code duplication here and in the other files. It would be good to reduce it by introducing a member method which might have the following signature and implementation:
```
void updateLeftRightScroll(int value)
{
   m_leftRightScroll->setValue(m_leftRightScroll->value() - value * 2 / 6 / m_zoomXLevels[m_zoomingXModel.value()]);
}
```
The code above would then simply call
```
updateLeftRightScroll(we->angleDelta().x())
```
and
```
updateLeftRightScroll(we->angleDelta().y())
```
respectively. That way if a change of implementation was needed it could simply be done in one place, i.e. in `updateLeftRightScroll`.

It would also be good to remove the consecutive divisions and replace them by one number.

Reply:
I think if we added a helper method, it would have to be implemented for each editor individually, since they all (except for PatternEditor) implement their own scrollbars. Is that what you were thinking?

I also agree it would be good to remove the finite zoom divisions and replace them with something like the SongEditor's smooth zoom slider, but I'll have to leave that for another PR since that would be a substantial amount of work.